dbms/src/Storages/LiveView/StorageBlocks.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <Storages/IStorage.h>
+
+
+namespace DB
+{
+
+class StorageBlocks : public IStorage
+{
+/* Storage based on the prepared streams that already contain data blocks.
+ * Used by Live Views to complete stored query based on the mergeable blocks.
+ */
+public:
+    StorageBlocks(const std::string & database_name_, const std::string & table_name_,
+        const ColumnsDescription & columns_, BlockInputStreams streams_,
+        QueryProcessingStage::Enum to_stage_)
+        : database_name(database_name_), table_name(table_name_), streams(streams_), to_stage(to_stage_)
+    {
+        setColumns(columns_);
+    }
+    static StoragePtr createStorage(std::string database_name, std::string table_name,
+        const ColumnsDescription columns, BlockInputStreams streams, QueryProcessingStage::Enum to_stage)
+    {
+        return std::make_shared<StorageBlocks>(std::move(database_name), std::move(table_name), std::move(columns), std::move(streams), to_stage);
```

Comment:
Actually, move does nothing here for `database_name`, `table_name` and `columns`, as they are passed to constructor via const reference. You need to pass them into constructor by value too (and move).

Reply:
You mean I should remove all the `const`s from the constructor so that they are passed by value and leave the move as is?

Reply:
Yes, remove `const &` and use `std::move` too in constructor.